be/src/exprs/json_functions.cpp

```diff
@@ -157,35 +157,42 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
         // we have to do some special treatment for the second elem of json path.
         // If the key is not found in json object, simdjson::NO_SUCH_FIELD would be returned.
         if (i == 1) {
+            if (obj.is_empty()) {
+                return Status::NotFound(fmt::format("unable to find key: {}", jsonpaths_to_string(jsonpath)));
+            }
+
             if (col == "*") {
                 // There should be no jsonpath for this pattern, $.*
-                return Status::InvalidArgument(fmt::format(
-                        "extracting * from root-object is not supported, the json path: {}", jsonpath[i].key));
+                return Status::InvalidArgument(
+                        fmt::format("extracting * from root-object is not supported, the json path: {}",
+                                    jsonpaths_to_string(jsonpath)));
             } else {
                 HANDLE_SIMDJSON_ERROR(obj.find_field_unordered(col).get(tvalue),
-                                      fmt::format("unable to find field: {}", col));
+                                      fmt::format("unable to find key: {}", jsonpaths_to_string(jsonpath)));
             }
         } else {
+            // There are always two patterns.
+            // 1. {"field_name": null}
+            // 2. {"field_name": {"field_type": data}}
+            // For pattern1, we just return null value.
+            // For pattern2, we get the first field of object as next value.
+
+            if (tvalue.is_null()) {
+                return Status::NotFound(fmt::format("unable to find key: {}", jsonpaths_to_string(jsonpath)));
+            }
+
+            if (tvalue.type() != simdjson::ondemand::json_type::object) {
+                return Status::NotFound(fmt::format("unable to find key: {}", jsonpaths_to_string(jsonpath)));
+            }
+
             if (col == "*") {
-                // There are always two patterns.
-                // 1. {"field_name": null}
-                // 2. {"field_name": {"field_type": data}}
-                // For pattern1, we just return null value.
-                // For pattern2, we get the first field of object as next value.
-                if (tvalue.is_null()) {
-                    return Status::NotFound("null value");
-                }
-                if (tvalue.type() != simdjson::ondemand::json_type::object) {
-                    return Status::InvalidArgument(fmt::format(
-                            "extracting * from non-object type is not supported, the json path: {}", jsonpath[i].key));
-                }
                 for (auto field : tvalue.get_object()) {
```

Comment:
will tvalue.get_object() return null if `$.a[*]` is not a JSON object, such as `{"a":1}`?

Reply:
Does  `$.a.*`  stand for any key name?

Reply:
> will tvalue.get_object() return null if `$.a[*]` is not a JSON object, such as `{"a":1}`?

Yes.

Reply:
> Does `$.a.*` stand for any key name?

`$.a.*` only works for `{"a": {"somekey": xxx}}`. The result is `{"somekey": xxx}`.
If the input is `{"a": 1}`, and the result would be null.